spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/ServerProperties.java

```diff
@@ -831,6 +832,11 @@ public void setSelectors(Integer selectors) {
 			 */
 			private boolean logLatency;
 
+			/**
+			 * Prefer IP address specified in X-Forwarded-For header.
+			 */
+			private boolean preferProxiedForAddress;
```

Comment:
I'd rather research if we couldn't apply this behaviour when `server.use-forward-headers` is set rather than requiring a new property.